test/webdriverio/test/actions_test.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as chai from 'chai';
+import {Key} from 'webdriverio';
+import {
+  checkActionPrecondition,
+  moveToToolboxCategory,
+  PAUSE_TIME,
+  setCurrentCursorNodeById,
+  tabNavigateToWorkspace,
+  testFileLocations,
+  testSetup,
+} from './test_setup.js';
+
+suite('Menus test', function () {
+  // Setting timeout to unlimited as these tests take longer time to run
+  this.timeout(0);
+
+  // Clear the workspace and load start blocks
+  setup(async function () {
+    this.browser = await testSetup(testFileLocations.BASE);
+    await this.browser.pause(PAUSE_TIME);
+  });
+
+  test('Menu action returns true normally', async function () {
+    // Navigate to draw_circle_1.
+    await tabNavigateToWorkspace(this.browser);
+    await setCurrentCursorNodeById(this.browser, 'draw_circle_1');
+    chai.assert.isTrue(
+      await checkActionPrecondition(this.browser, 'menu'),
+      'The menu should be openable on a block',
+    );
+  });
+
+  test('Menu action returns true in the toolbox', async function () {
+    // Navigate to draw_circle_1.
+    await tabNavigateToWorkspace(this.browser);
+    await setCurrentCursorNodeById(this.browser, 'draw_circle_1');
+    // Navigate to a toolbox category
+    await moveToToolboxCategory(this.browser, 'Functions');
+    // Move to flyout.
+    await this.browser.keys(Key.ArrowRight);
+
+    chai.assert.isTrue(
+      await checkActionPrecondition(this.browser, 'menu'),
```

Comment:
Is it possible to actually open the menu and verify that it's open rather than checking the precondition? That would be a lot closer to what the user experiences (and thus perhaps a bit more robust against implementation changes and refactors that don't change the end user behavior).